tests/disk.rs

```diff
@@ -0,0 +1,62 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[test]
+#[cfg(all(feature = "system", feature = "disk"))]
+fn test_disks() {
+    if sysinfo::IS_SUPPORTED_SYSTEM {
+        let s = sysinfo::System::new_all();
+        // If we don't have any physical core present, it's very likely that we're inside a VM...
+        if s.physical_core_count().unwrap_or_default() > 0 {
+            let mut disks = sysinfo::Disks::new();
+            assert!(disks.list().is_empty());
+            disks.refresh_list();
+            assert!(!disks.list().is_empty());
+        }
+    }
+}
+
+#[test]
+#[cfg(feature = "disk")]
+fn test_disks_usage() {
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    let s = sysinfo::System::new_all();
+
+    if !sysinfo::IS_SUPPORTED_SYSTEM || s.physical_core_count().unwrap_or_default() == 0 {
```

Comment:
Seems like it's not enough...

Just had an idea though. Can you create file in `Path::new(std::env::var_os("CARGO_TARGET_DIR").expect("missing CARGO_TARGET_DIR env var")).join("disk_usage_test.tmp")` and write into it?

Reply:
Yeah, seems like the CI runner has 2 cores.

I tried it with my test branch and CARGO_TARGET_DIR doesn't seem to be defined for some odd reason: https://github.com/samin-cf/sysinfo/actions/runs/11672513236/job/32501288423

Reply:
Try `CARGO_TARGET_TMPDIR` as fallback?

Reply:
Weird. No `CARGO` environment variables are defined: https://github.com/samin-cf/sysinfo/actions/runs/11672628189/job/32501653467

Reply:
There are a few, but not the ones we're looking for. Let's try at compile time with `env!` then?

Reply:
`CARGO_TARGET_DIR` isn't defined during compile time, but `CARGO_TARGET_TMPDIR` is. Still no luck: https://github.com/samin-cf/sysinfo/actions/runs/11672913856/job/32502524730

Reply:
I didn't expect it to be such a nightmare. ^^'

Then just put the file in `target/test_file.tmp` and let's hope for the best this time...

Reply:
😢 https://github.com/samin-cf/sysinfo/actions/runs/11673224654/job/32503482984#step:4:124

With the build just before the one I linked, the file was created in was `target/tmp` and that also failed.

I am finding it surprising no one has run into this. Either that or something is up with the implementation (but it works locally 🙃)

Reply:
I think the best course of action here is disabling in CI. Since we saw what env variables were set, we can disable it on linux if we're in github CI.

Reply:
Thanks for the help along the way! Quite the journey lol